src/main/java/ToDoApplication/DAO/BaseDao.java

```diff
@@ -0,0 +1,52 @@
+package ToDoApplication.DAO;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+abstract class BaseDao {
+
+
+    protected void closeResource(Connection connection, PreparedStatement preparedStatement) {
+        if (connection != null) {
+            try {
+                connection.close();
+            } catch (SQLException e) {
+                throw new RuntimeException(e.getMessage(), e);
+            }
+        }
+        if (preparedStatement != null) {
+            try {
+                preparedStatement.close();
+            } catch (SQLException e) {
+                throw new RuntimeException(e.getMessage(), e);
+            }
+        }
+    }
+
+    protected void closeResource(Connection connection, PreparedStatement preparedStatement, ResultSet resultSet){
+        if (connection != null){
```

Comment:
You have here repetitive code. You use same code for closing Connection and PreparedStatement objects in both methods. I think it would be more efficient to call closeResource(connection, preparedStatement) method, instead of writing exactly the same code.